crates/bevy_input/src/gamepad.rs

```diff
@@ -1127,102 +983,256 @@ impl ButtonAxisSettings {
 /// Whenever a [`Gamepad`] connects or disconnects, an information gets printed to the console using the [`info!`] macro.
```

Comment:
Nit: This seems to do a bit more than just printing debug info as it handles registering/deregistering gamepads, along with other stuff. Maybe that should be included in the doc comment?